src/paas_charm/utils.py

```diff
@@ -103,3 +103,24 @@ def config_get_with_secret(
     if secret_id is None:
         return None
     return charm.model.get_secret(id=typing.cast(str, secret_id))
+
+
+def get_missing_non_optional_configs(charm: ops.CharmBase) -> list[str] | None:
+    """Get a list of missing non-optional config options.
+
+    Args:
+        charm: The charm instance.
+
+    Returns:
+        Returns the list of missing non-optional config options,
+        if none exists returns None.
+    """
+    metadata = _config_metadata(pathlib.Path(os.getcwd()))
+    missing_options = []
+    for option in metadata["options"]:
+        if metadata["options"][option].get("optional") is False:
+            if not charm.config.get(option):
```

Comment:
```suggestion
            if charm.config.get(option) is None:
```
I think using `is None` would be better here since some configurations may contain `0` or `""`, which are falsey.